.github/workflows/documentation.yml

```diff
@@ -0,0 +1,50 @@
+name: documentation
+
+on:
+  pull_request:
+    branches: [documentation]
+  push:
+    branches: [documentation]
+
+jobs:
+ checks:
+   if: github.event_name != 'push'
+   runs-on: ubuntu-latest
+   steps:
+     - uses: actions/checkout@v1
+     - uses: actions/setup-node@v1
+       with:
+         node-version: '12.x'
+         run: |
+          npm ci
+          npm run build
+ gh-release:
+   if: github.event_name != 'pull_request'
+   runs-on: ubuntu-latest
+   steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - name: Add key
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: |
+        mkdir -p ~/.ssh
+        ssh-keyscan github.com >> ~/.ssh/known_hosts
+        echo "${{ secrets.SERVER_DEPLOY_KEY }}" > ~/.ssh/id_rsa
+        chmod 600 ~/.ssh/id_rsa
+        cat <<EOT >> ~/.ssh/config
+        Host github.com
+        HostName github.com
+        IdentityFile ~/.ssh/id_rsa
+        EOT
```

Comment:
The "Add key" action seems quite hacky. Is this really necessary? You already have the deploy key (and others) in the secrets.
I am not sure whether this is helpful, but take a look at this: https://github.com/marketplace/actions/deploy-to-github-pages

Reply:
I considered the action you refer to @marikaner, but it's a 3rd party one and not certified by GitHub. This led to:
- security considerations that forced me to organize key management in a different way
- the key is needed for the sole purpose of Docusurus using git to clone the a repository and doing other git magic to deploy. That's also a reason why the gh action mentioned above wouldn't probably help.
- The way is not that hacky. Just wiring .ssh for Git to do its job. You'll find it on most servers and dev machines. The `Deploy` key actually exists for those "deploy" machines from which you can do deployment. In our case, we store the `Deploy` public key where it should be and add inject its private counterpart via Github Secrets.
- Everything else it laying on shoulders of `docusaurus deploy` which is good to my opinion, because it's maintainable and consistent.

I admit hiding some part of this into containerized action would be nice, but that also leads to needless overhead at the moment. Do you think we should invest in it at the moment?
I'm happy if someone more experienced contributes a more beautiful solution for this. I had a meeting with Florian Wilhelm, but he also hasn't dealt with such cases.

Reply:
To avoid confusion, the `Deploy key` was added by me + one additional secret.

Reply:
Ok, so I meant take a look at others for inspiration rather than using an external action. Now, I played around a bit and came to the conclusion that using a github token might be a simpler way. At least it has less steps. I tried to use the default `GITHUB_TOKEN` but did not succeed with this. I tried and confirmed that the following works:
```
- name: Release to GitHub Pages
      env:
        GITHUB_TOKEN: ${{ secrets.GH_WRITE_TOKEN }}
        GIT_USER: ${{ secrets.GH_WRITE_USER }}
      run: |
        git config --global user.name ${GIT_USER}	
        git config --global user.email 'irrelevant@github.com'
        echo "machine github.com login ${GIT_USER} password ${GITHUB_TOKEN}" > ~/.netrc
        npm ci
        npx docusaurus deploy
```

Reply:
Thanks for putting effort into it. These are equivalent solutions. Both use a feature of injecting credentials to make needed git commands to Linux env, but differently. 
- With `token` there is a risk of revocation by a team member that changes teams for example. Leaking it has similar implications to the private key.
- With a pair of `Deploy public/private` keys revocation is not an issue. Also bad if it's leaked but the fix is the same a with the personal token.

I don't have strong opinions about it. Either option works with me. The private key can be probably converted into a one-liner for beautification but I would be reluctant to do it at this stage.

I can close this pull request so that you can contribute your version. My only goal is to have the initial contribution pipeline as fast a possible with meaningful effort and quality.

Reply:
We'll definitely have a couple of iterations brushing it up to add linting to JS part of the site, adding tests for react components, running Markdown linting, etc. It all not relevant now compared to actual useful information we proved for customers in the form of code examples and concepts making them use SDK efficiently.